lib/src/test/java/io/ably/lib/test/rest/RestAuthAttributeTest.java

```diff
@@ -366,4 +366,67 @@ public void auth_authorise_only_force() {
 			fail("auth_custom_options_authorise: Unexpected exception");
 		}
 	}
+
+	/**
+	 * Verify if the {@link AuthOptions#queryTime} is true, it will obtain
+	 * the server time once and persist the offset from the local clock.
+	 * All future token requests generated directly or indirectly via a call
+	 * to authorise will not obtain the server time,
+	 * but instead use the local clock offset to calculate the server time.
+	 * <p>
+	 * Spec: RSA10k
+	 * </p>
+	 */
+	private int counter = 0;
+
+	@Test
+	public void auth_authorise_query_time() {
+		try {
+			/* init custom Ably client with counter for time() method */
+			Setup.TestVars testVars = Setup.getTestVars();
+			ClientOptions opts = testVars.createOptions();
+			AblyRest ablyRest = new AblyRest(opts) {
+				@Override
+				public long time() throws AblyException {
+					counter++;
+					return super.time();
+				}
+			};
+
+			/* init custom auth options for obtaining the server time */
+			AuthOptions authOptions = new AuthOptions();
+			authOptions.force = true;
+			authOptions.queryTime = true;
+			authOptions.key = testVars.keys[0].keyStr;
+
+			/* authorise with custom auth options */
+			TokenDetails tokenDetails1 = ablyRest.auth.authorise(authOptions, null);
+
+			/* Verify that, token isn't null and method time() invoked once */
+			assertNotNull("Expected token value", tokenDetails1.token);
+			assertEquals("Excepted one invoke", counter, 1);
+
+			/* re-authorise with same custom auth options */
+			TokenDetails tokenDetails2 = ablyRest.auth.authorise(authOptions, null);
+
+			/* Verify that,
+			 * new token isn't null, tokens aren't equals and
+			 * method time() not invoked again */
+			assertNotNull("Expected token value", tokenDetails2.token);
+			assertNotEquals(tokenDetails1.token, tokenDetails2.token);
+			assertEquals("Excepted one invoke", counter, 1);
+
+			/* discard the cached local clock offset and
+			 * re-authorise with same custom auth options */
+			ablyRest.auth.discardTimeOffset();
+			TokenDetails tokenDetails = ablyRest.auth.authorise(authOptions, null);
+
+			/* Verify that, issued token isn't null and method time() invoked twice */
+			assertNotNull("Expected token value", tokenDetails.token);
+			assertEquals("Excepted one invoke", counter, 2);
```

Comment:
@mattheworiordan So, for testing by intentional offset, I should make persisted `timeOffset` field public for changing it in a test. But there is no documented API for this. I can get to access private field via reflection. It'll be correct?